cli/commands/info/index.ts

```diff
@@ -0,0 +1,116 @@
+import { formatIpfsData, GetFromIpfs, IpfsMetadata } from "../../utils/ipfs/get.from.ipfs";
+import { providers } from "ethers";
+import { GetTransactionReceipt } from "../../utils/get.transaction.receipt";
+import { flatten } from "lodash";
+
+const SECONDS_IN_DAY = 86000;
+const DEFAULT_DAYS_TO_SCAN = 30;
+
+export default function provideInfo(
+    agentId: string,
+    args: any,
+    ethersAgentRegistryProvider: providers.JsonRpcProvider,
+    agentRegistry: AgentRegistry,
+    agentRegistryContractAddress: string,
+    getFromIpfs: GetFromIpfs,
+    getTransactionReceipt: GetTransactionReceipt,
+): CommandHandler {
+    assertExists(args, 'args')
+    assertExists(ethersAgentRegistryProvider, 'ethersAgentRegistryProvider')
+    assertExists(agentRegistry, 'agentRegistry')
+    assertExists(agentRegistryContractAddress, 'agentRegistryContractAddress')
+    assertExists(getFromIpfs, 'getFromIpfs')
+    assertExists(getTransactionReceipt, 'getTransactionReceipt')
+
+    return async function info(daysToScan = DEFAULT_DAYS_TO_SCAN) {
+        const finalAgentId = args.agentId ? args.agentId : agentId;
+
+        assertIsNonEmptyString(finalAgentId, 'agentId');
+
+        const agent = await agentRegistry.getAgent(finalAgentId);
+        const currentState = await agentRegistry.isEnabled(finalAgentId) as boolean
+        const ipfsMetaHash = agent.metadata;
+
+
+        const ipfsData = await getFromIpfs(ipfsMetaHash)
+        printIpfsMetaData(ipfsData,currentState)
+
+        console.log(`Recent Activity (Last ${daysToScan} days): \n`);
+
+        const blockEventTopicFilters = AGENT_REGESTRY_EVENT_FRAGMENTS
+            .filter(fragment => isRelevantSmartContractEvent(fragment.name))
+            .map(eventFragment => getTopicHashFromEventName(eventFragment.name as StateChangeContractEvent)) as string[];
+
+        const latestBlockNumber = await ethersAgentRegistryProvider.getBlockNumber();
+
+        const network = await ethersAgentRegistryProvider.getNetwork();
+        const { chainId } = network;
+        const { blockTimeInSeconds } = getBlockChainNetworkConfig(chainId);
+
+        console.log(`Days to scan is: ${daysToScan}, Block to subtract is: ${(daysToScan * SECONDS_IN_DAY)/(blockTimeInSeconds)}`)
+        const endingBlock = Math.floor(latestBlockNumber - ((daysToScan * SECONDS_IN_DAY)/(blockTimeInSeconds)));
+
+        const increment = 1000;
+        let startingBlock = latestBlockNumber - (increment * 5);
+
+        const getAgentLogs = async (startBlock: number, endBlock: number) => {
+            return await ethersAgentRegistryProvider.getLogs({
+                address: agentRegistryContractAddress,
+                fromBlock: startBlock,
+                toBlock: endBlock,
+                topics: [[...blockEventTopicFilters], null] // This assumes that each smart contract event is indexed so that the first topic is always an event of interest
```

Comment:
we can further narrow down the getLogs query by specifying event parameters that are marked as `indexed` i.e. the agentId. the indexed parameters become topics on the event log. see here for more details: https://eth.wiki/json-rpc/API#a-note-on-specifying-topic-filters

Reply:
Used polyscans api to handle narrowing down the logs to `indexed`